sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1966,7 +1966,7 @@ case class ElementAt(left: Expression, right: Expression)
   }
 
   override def nullable: Boolean = left.dataType match {
-    case _: ArrayType => computeNullabilityFromArray(left, right)
+    case _: ArrayType => computeNullabilityFromArray(left, right, isOneBasedIndex = true)
```

Comment:
how about `computeNullabilityFromArray(left, Subtract(right, Literal(1)))`?

Reply:
what if it's a negative number like -1. -1 means the last one from right to left.

Reply:
```
_FUNC_(array, index) - Returns element of array at given (1-based) index. If index < 0,
      accesses elements from the last to the first. Returns NULL if the index exceeds the length
      of the array.
```

Reply:
Got it. Then `isOneBasedIndex` is a misleading name. Maybe the parameter should be `normalizeIndex: (Int, Int) => Int = _._2`, which takes the array length and the index, and return the normalized 0-based non-negative index.